charts/internal/seed-controlplane/charts/remedy-controller-azure/templates/deployment.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: remedy-controller-azure
+  namespace: {{ .Release.Namespace }}
+  labels:
+    garden.sapcloud.io/role: controlplane
+    app: remedy-controller-azure
+spec:
+  revisionHistoryLimit: 0
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: remedy-controller-azure
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-remedy-controller-azure-config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        garden.sapcloud.io/role: controlplane
+        app: remedy-controller-azure
+        networking.gardener.cloud/to-dns: allowed
+        networking.gardener.cloud/to-public-networks: allowed
+        networking.gardener.cloud/to-private-networks: allowed
+        networking.gardener.cloud/to-seed-apiserver: allowed
+        networking.gardener.cloud/to-shoot-apiserver: allowed
+        networking.gardener.cloud/from-prometheus: allowed
+    spec:
+      serviceAccountName: remedy-controller-azure
+      tolerations:
```

Comment:
Why do we need this toleration? Is the remedy controller that important that we also schedule it on nodes with a `NoExecute` taint? If yes do we need maybe a `tolerationSeconds` in case the taint after the pod is already running on the pod?

Reply:
That's a good question, I am not sure myself. I noticed that this toleration is present for the `cloud-controller-manager` for Azure and other platforms, as well as well as on the AWS LB readvertiser, and I thought the remedy controller was of comparable importance, but this was just a guess and so perhaps is wrong. Perhaps the interesting question is, why anything that runs in the seed cluster, including the components above, would need this toleration? I would understand it for system components running in the shoot, e.g. the CSI node driver. I also remember this is explicitly stated in the CCM documentation, if CCM runs on the cluster nodes (CCM itself removes the taint when a node is added, so if CCM doesn't tolerate the taint, it can't run in order to do this), but in the Gardener case, I believe this doesn't apply? If there is a good reason, perhaps it applies to the remedy controller. If not, perhaps we should remove it from other places as well.

Reply:
Ok then we should maybe also remove the toleration from the other components which @stoyanr mentioned. Thanks for clarification.

Reply:
OK, I will remove it and if you don't mind, will fix Azure CCM as well. Should I open PRs for the other providers with this change?

Reply:
Thanks, if you have the time, sure. But I think it's not critical, so go ahead as it fits your schedule.